Sources/DockProgress/DockProgress.swift

```diff
@@ -1,11 +1,25 @@
 import Cocoa
+import CoreVideo
 
 @MainActor
 public enum DockProgress {
-	private static var previousProgress: Double = 0
 	private static var progressObserver: NSKeyValueObservation?
 	private static var finishedObserver: NSKeyValueObservation?
 
+    private static var t = 0.0
+    private static var displayLinkObserver = DisplayLinkObserver { (displayLinkObserver, refreshPeriod) in
+        let speed = 1.0
+        t += speed * refreshPeriod
+        if (animatedProgress - progress).magnitude <= 0.01 {
+            animatedProgress = progress
+            t = 0
+            displayLinkObserver.stop()
+        } else {
+            animatedProgress = Easing.lerp(animatedProgress, progress, Easing.easeInOut(t));
+        }
+        updateDockIcon()
+    }
```

Comment:
Let's just wrap it in `DispatchQueue.main.async` as the `Task` has limited concurrency.

```suggestion
	private static var displayLinkObserver = DisplayLinkObserver { (displayLinkObserver, refreshPeriod) in
		DispatchQueue.main.async {
			let speed = 1.0
			t += speed * refreshPeriod
			if (animatedProgress - progress).magnitude <= 0.01 {
				animatedProgress = progress
				t = 0
				displayLinkObserver.stop()
			} else {
				animatedProgress = Easing.lerp(animatedProgress, progress, Easing.easeInOut(t));
			}
			updateDockIcon()
		}
	}
```